src/webgl/p5.RendererGL.js

```diff
@@ -1874,6 +1905,85 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
     this.textures.set(src, tex);
     return tex;
   }
+  /*
+    *  used in imageLight,
+    *  To create a blurry image from the input non blurry img,
+    *  Add it to the blurryTexture map,
+    *  Returns the blurry image
+   */
+  getBlurryTexture(input){
+    // if one already exists for a given input image
+    if(this.diffusedTextures.get(input)!=null){
+      return this.diffusedTextures.get(input);
+    }
+    // if not, then create one
+    let newGraphic; // maybe switch to framebuffer
+    // draw the blurry image
+    // set the shader on the graphic and set the shader on the image
+    // this._renderer._applyTextProperties(img, newGraphic);
+    // make small width, hardcode to 200px
+    let smallWidth = 200;
+    //let smallWidth = input.width;
+    let width = smallWidth;
+    let height = Math.floor(smallWidth * (input.height / input.width));
+    newGraphic = createGraphics(width, height, WEBGL);
+    // create graphics is like making a new sketch, all functions on main
+    // sketch it would be available on graphics
+    let irradiance = newGraphic.createShader(
+      defaultShaders.imageLightVert,
+      defaultShaders.imageLightDiffusedFrag
+    );
+    newGraphic.shader(irradiance);
+    irradiance.setUniform('environmentMap', input);
+    newGraphic.noStroke();
+    newGraphic.rectMode(newGraphic.CENTER);
+    newGraphic.rect(0, 0, newGraphic.width, newGraphic.height);
+    this.diffusedTextures.set(input, newGraphic);
+    return newGraphic;
+  }
+
+  /*
+   *  used in imageLight,
+   *  To create a texture from the input non blurry image,
+   *  Creating 8 different levels of textures according to different sizes and atoring them in `levels` array
```

Comment:
Maybe worth mentioning that they're different roughness levels too. The rougher it is, the smaller the size, because rougher surfaces are blurrier/less detailed